pytorch_lightning/metrics/classification/helpers.py

```diff
@@ -410,6 +418,9 @@ def _input_format_classification(
     if preds.dtype == torch.float16:
         preds = preds.float()
 
+    # Let threshold default to 0.5 if not set
+    threshold = 0.5 if threshold is None else threshold
```

Comment:
how it happens that `threshold is None` while arg default is 0.5?

Reply:
So same as above, the new default is now `None`. This was suggested by @rohitgr7, because for multi-class inputs the `threshold` parameter does not apply, so this makes it less confusing for those cases

Reply:
I may see a reason to have none for metrics where it does not apply to keep the same API, but on the other hand, this not using default value if possible make it just less readable...

Reply:
Ok, I agree with this - will revert back to 0.5 as default